File: xor/xor.py
import sys

XOR_KEY = 0xFF
CHUNK_SIZE = 2048

def main():
    file_name = 'gta3.img'
    with open(file_name, 'rb') as infile, \
        open(file_name + '-xor', 'wb') as outfile:
        for chunk in iter(lambda: infile.read(CHUNK_SIZE), b''):
            xored = bytearray(chunk)
            for i, _ in enumerate(xored):
                xored[i] ^= XOR_KEY
            outfile.write(xored)


if __name__ == '__main__':
    main()

File: xor/xor.js
const fs = require("fs");

const XOR_KEY = 0xFF;
const CHUNK_SIZE = 2048;

let PATH = "gta3.img"
let OUTPUT_PATH = PATH + "-xor";

let readStream = fs.createReadStream(PATH, { highWaterMark: CHUNK_SIZE });
let writeStream = fs.createWriteStream(OUTPUT_PATH, { flags: "w+" });

readStream.on("data", (chunk) => {
	for (let i = 0; i < chunk.length; i++)
		chunk[i] ^= XOR_KEY;
	writeStream.write(chunk);
}).on("end", () => { writeStream.end(); });
